molo/profiles/forms.py

```diff
@@ -95,6 +147,37 @@ class Meta:
         model = UserProfile
         fields = ['alias', 'date_of_birth', 'mobile_number']
 
+    def clean_alias(self):
+        site = Site.objects.get(is_default_site=True)
+        settings = SettingsProxy(site)
+        profile_settings = settings['profiles']['UserProfilesSettings']
+
+        invalid_msg = ''
+
+        if getattr(profile_settings, 'prevent_email_in_username', False) \
+                and getattr(profile_settings, 'prevent_number_in_username',
+                            False):
+            invalid_msg = 'phone number or email address'
+
+        elif getattr(profile_settings, 'prevent_number_in_username', False):
+            invalid_msg = 'phone number'
+
+        elif getattr(profile_settings, 'prevent_email_in_username', False):
+            invalid_msg = 'email address'
```

Comment:
Everything up to here is a repetition of what's happening in `clean_username`. I would factor out the repeated code into, for example, `get_validation_msg_fragment`.

Then both `clean_alias` and `clean_username` can start with something like:
```python
validation_msg_fragment = get_validation_msg_fragment()
```
which you can then use when you're formatting the text for your ValidationError.